agent/xds/listeners_ingress.go

```diff
@@ -134,7 +167,7 @@ func (s *ResourceGenerator) makeIngressGatewayListeners(address string, cfgSnap
 
 			// See if there are other services that didn't have specific SNI-matching
 			// filter chains. If so add a default filterchain to serve them.
-			if len(sniFilterChains) < len(upstreams) {
+			if len(sniFilterChains) < len(upstreams) && !isAPIGatewayWithTLS {
```

Comment:
we bypass this as it's just creating a default filter chain which we already did inside of the new override function

Reply:
I think if it's worth calling out this way in a PR, you could probably safely add a comment in the code explaining the same.